src/redux/saga/fbSignInSaga.ts

```diff
@@ -0,0 +1,56 @@
+import { AccessToken, LoginManager } from 'react-native-fbsdk-next';
+import auth from '@react-native-firebase/auth';
+import { AccessTokenMap } from 'react-native-fbsdk-next/src/FBAccessToken';
+import { createErrorAlertMessageHelper } from '../../helpers/createErrorAlertMessageHelper';
+import { FirebaseDatabaseTypes } from '@react-native-firebase/database';
+import { createReferenceHelper } from '../../helpers/createReferenceHelper';
+import { put } from 'redux-saga/effects';
+import { successSignInAction } from '../actions/authActions/successSignInAction';
+
+export function* fbSignInSaga() {
+  try {
+    // Attempt login with permissions
+    yield LoginManager.logInWithPermissions(['public_profile', 'email']);
+
+    // Once signed in, get the users AccessToken
+    const dataToken: AccessTokenMap = yield AccessToken.getCurrentAccessToken();
+    const accessToken = dataToken.accessToken;
+    // const user = profileRequest.callback();
+    if (!accessToken) {
+      createErrorAlertMessageHelper(
+        'Something went wrong obtaining access accessToken',
+      );
+    }
+
+    console.log('dataToken', dataToken);
+    // Create a Firebase credential with the AccessToken
+    const facebookCredential = auth.FacebookAuthProvider.credential(
+      dataToken.accessToken,
+    );
+    console.log('facebookCredential   ', facebookCredential);
+
+    // Sign-in the user with the credential
```

Comment:
I saw a similar code in google sign in, try to do a separate saga that will get params and will save the token and other necessary things, and will be called here and in google, because right now it looks like copy past from google.

Reply:
doing it